src/evm/P2pInteraction.ts

```diff
@@ -0,0 +1,32 @@
+import P2pSigner from "./P2pSigner";
+import { AStateMachine } from "@typechain-types";
+import P2pEventHooks from "@/P2pEventHooks";
+
+/**
+ * Represents a P2P interaction with a state machine contract
+ */
+export class P2pInteraction<T extends AStateMachine> {
+    p2pContractInstance: T;
```

Comment:
Need a better name for this - this is the dumbest name I came up with in the entire code base and this is the only thing the user of the SDK sees

Reply:
renamed to StateChannelSession